practice-app/server/test/enroll_lesson.test.js

```diff
@@ -0,0 +1,209 @@
+        .post(enrollUrl)
+        .send({
+          user_id: "id",  
+          lesson_id: lesson_ids[0]
+         })
+        .expect(400)
+        .end(done);
+    });
+  
+    it('should return validation error if the lesson_id is invalid in body', (done) => {
+      request(app)
+        .post(enrollUrl)
+        .send({
+          user_id: user_ids[0],  
+          lesson_id: "55444645161"
+         })
+        .expect(400)
+        .end(done);
+    });
+  
+    it('should return not found error if the user with the given user_id does not exist', (done) => {
+      request(app)
+        .post(enrollUrl)
+        .send({
+          user_id: invalid_ids[0],  
+          lesson_id: lesson_ids[0]
+         })
+        .expect(404)
+        .end(done);
+    });
+  
+    it('should return not found error if the lesson with the given lesson_id does not exist', (done) => {
+      request(app)
+        .post(enrollUrl)
+        .send({
+          user_id: user_ids[0],  
+          lesson_id: invalid_ids[1]
+         })
+        .expect(404)
+        .end(done);
+    });
+  
+    it('should return not found error if the user with the given user_id does take the lesson with the given lesson_id', (done) => {
+      request(app)
+        .post(enrollUrl)
+        .send({
+          user_id: user_ids[0],  
+          lesson_id: lesson_ids[0]
+         })
+        .expect(404)
+        .end(done);
+    });
+  
+    it('should return the user with the given user_id with updated enrolledLessons list and added lesson with the given lesson_id', (done) => {
+  
+      request(app)
+        .post(enrollUrl)
+        .send({ user_id: user_ids[1], lesson_id: lesson_ids[0] })
+        .expect((res) => {
+          expect(res.body.user).not.toBeNull();
+          expect(res.body.user).not.toBeUndefined();
+          expect(res.body.user.email).not.toBeNull();
+          expect(res.body.user.email).not.toBeUndefined();
+          expect(res.body.user.name).not.toBeNull();
+          expect(res.body.user.name).not.toBeUndefined();
+          expect(res.body.user.enrolledLessons).not.toBeNull();
+          expect(res.body.user.enrolledLessons).not.toBeUndefined();
+          expect(res.body.user.enrolledLessons).toHaveLength(2);
+          expect(res.body.lesson).not.toBeNull();
+          expect(res.body.lesson).not.toBeUndefined();
+          expect(res.body.lesson.id).not.toBeNull();
+          expect(res.body.lesson.id).not.toBeUndefined();
+          const firstLesson = res.body.user.enrolledLessons[0];
+          expect(firstLesson).not.toBeNull();
+          expect(firstLesson).not.toBeUndefined();
+          expect(res.statusCode).toBe(200);
+        })
+        .end(done);
+    });
+    
+    it('should return the user with the given user_id with remaining enrolledLessons list and removed lesson with the given lesson_id', (done) => {
+  
+      request(app)
+        .post(enrollUrl)
+        .send({ user_id: user_ids[1], lesson_id: lesson_ids[3] })
```

Comment:
I am not sure but I think there is no lesson with the id lesson_ids[3], so last test would give an error with status code 400.

Reply:
I think this resolved now.

Reply:
Yes, thank you.